pkg/controller/jobframework/interface.go

```diff
@@ -58,10 +58,14 @@ func ParentWorkloadName(job GenericJob) string {
 	return job.Object().GetAnnotations()[ParentWorkloadAnnotation]
 }
 
-func QueueName(job GenericJob) string {
-	if queueLabel := job.Object().GetLabels()[QueueLabel]; queueLabel != "" {
+func QueueNameForGenericJob(job GenericJob) string {
```

Comment:
Since most of the functions in this package operate on GenericJob, I would instead let the existing function keep its name, and add QueueNameForObject.

It will also reduce the diff size.

Reply:
That makes sense.

Reply:
done.